### ISSUE TITLE
Menu: steals focus from input component

### ISSUE THREAD

**Issue by milansie** [primefaces/primefaces#13168]
### Describe the bug

Hi @melloware,

You will probably start to hate me :), but we have discovered another issue. This time it might be more of a discussion point - whether this is a bug, a feature, or just a misunderstanding.

Let me describe the problematic behaviour: There are some input fields and a menu/menubar (with toggleEvent on hover) on the page. When you enter any input field and start typing, then move the mouse (accidentally or intentionally) over any menu item, the menu gains focus and you cannot continue typing.

Applications with click-activated menus work similarly, but clicking on the menu is a deliberate user choice. Hovering over the menu is more likely to be accidental (e.g., when moving the mouse out of the way).

What should we do about this? I understand that these changes were made to fulfil WCAG criteria (I assume). But is this the correct behaviour? When checking other major websites, they either have clickable menus, or if anything is activated by hover, it doesn't steal focus from the input field. I think PrimeFaces should handle it similarly, but the decision is yours. :)

### Reproducer

https://www.primefaces.org/showcase/ui/menu/menubar.xhtml

just start typing into search field and then move mouse over the menu

### Expected behavior

who knows, I personally think, that focus should stay in input field, when menu is controlled by hovering the mouse

### PrimeFaces edition

Community

### PrimeFaces version

14.0.x

### Theme

all

### JSF implementation

All

### JSF version

all

### Java version

all

### Browser(s)

all

**Comment by milansie** [comment]
Hm, i have tested that on 15.0.0-SNAPSHOT and still facing this issue. Tested that on Firefox and Chrome-like browser on Linux.

**Comment by melloware** [comment]
hmmm can you put together a small test that shows it? I just used the 15.0.0 showcase above and couldn't reproduce it?

**Comment by milansie** [comment]
Could you reproduce that issue on PF showcase here? https://www.primefaces.org/showcase/ui/menu/menubar.xhtml

**Comment by melloware** [comment]
oh i see its only specifically the Search BOX INSIDE the Menubar i created an InputText that lives outside the menubar.

**Comment by milansie** [comment]
No, I created other input text outside the menubar, and the issue occurs on these input texts too
```
        <div class="card">
            <h:form>
                <p:growl id="messages"/>
                <p:inputText placeholder="Search"/>
                <h:inputText label="aaa" />

                <p:menubar>
                    <p:submenu label="File" icon="pi pi-fw pi-file">
```

**Comment by melloware** [comment]
MonkeyPatch: 

```js
if (PrimeFaces.widget.TieredMenu) {
  PrimeFaces.widget.TieredMenu.prototype.bindFocusEvents = function() {
    var $this = this;

    // Make first menu item focusable
    var firstLink = this.links.filter(':not([disabled])').first();
    firstLink.attr("tabindex", $this.tabIndex);
    this.resetFocus(true);
    firstLink.removeClass('ui-state-hover ui-state-active');

    var focusOnClick = this.cfg.toggleEvent === 'click';
    var linkEvents = "mouseenter.tieredFocus" + (focusOnClick ? " click.tieredFocus" : "");

    // Bind mouse/click events to manage focus
    this.links.on(linkEvents, function() {
      var $link = $(this),
        $menuitem = $link.parent();
      $this.deactivate($menuitem);
      if (focusOnClick) {
        $link.trigger('focus');
      } else {
        $this.highlight($menuitem);
      }
    }).on("focusin.tieredFocus", function() {
      var menuitem = $(this).parent();
      $this.highlight(menuitem);
    });
  }
}
```

**Comment by milansie** [comment]
I think this change doesn't help. 
I've just tested it, and the menu still steals the focus. Through bindHoverModeEvents -> $this.activate(menuitem) 

there is
```
       // focus the menu item when activated
        this.focus(menuitem.children('a.ui-menuitem-link'));
```

**Comment by milansie** [comment]
I would prefer to call this only through clickable (keyboard) navigation, rather than on hover. 
However, I am uncertain whether accessibility tools like screen readers function when an item is not focused. For accessibility purposes, it might be necessary for the menu item to maintain focus even when the mouse hovers over it.

**Comment by melloware** [comment]
I will look further.

**Comment by melloware** [comment]
OK check PR had to make more changes.  Basically i need to know whether its a mouse event or a keyboard/click event to determine whether to force the focus or not.

**Comment by milansie** [comment]
At first glance, it appears to work well. I will do some tests in the real application tomorrow and keep you updated. Thank you!

**Comment by melloware** [comment]
Thank you! Let me know I do think this is a good improvement that still keeps the accessibility requirements.

**Comment by milansie** [comment]
Hi @melloware,
I have done some testing in a real application, and everything looks perfect now. 
Excellent work, as always. Thank you very much.

**Comment by nikolica666** [comment]
I have very similar behaviour in 15.0.13 when i use <p:menu> and <p:inputText> hovering over menu loses inputText focus. Let's say that menu is your main navigation menu. and you wanna focus inputText straight after navigation. the moment you "touch" your mouse after clicking on desired view, inputText focus is lost.

I solved it switching to <p:tieredMenu> just wanted to comment and not open another issue
